examples/Misc/UploadMediaBundle.php

```diff
@@ -0,0 +1,134 @@
+use Google\Ads\GoogleAds\V2\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V2\Resources\MediaBundle;
+use Google\Ads\GoogleAds\V2\Resources\MediaFile;
+use Google\Ads\GoogleAds\V2\Services\MediaFileOperation;
+use Google\ApiCore\ApiException;
+use Google\Protobuf\StringValue;
+
+/** This example uploads an HTML5 zip file as a media bundle. */
+class UploadMediaBundle
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    const BUNDLE_URL = 'https://goo.gl/9Y7qI2';
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())
+            ->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
+    {
+        // Creates an HTML5 zip file media bundle content.
+        $bundleContent = file_get_contents(self::BUNDLE_URL);
+
+        // Creates a media file.
+        $mediaFile = new MediaFile([
+            'name' => new StringValue(['value' => 'Ad Media Bundle']),
+            'type' => MediaType::MEDIA_BUNDLE,
+            'source_url' => new StringValue(['value' => self::BUNDLE_URL]),
```

Comment:
Interesting that we need to provide BUNDLE_URL twice here.
source_url is a required field?

Reply:
Good catch, this field is 'Only used for media of type AUDIO and IMAGE' so not needed here.